spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/cloudmap/properties/CloudMapProperties.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.cloudmap.properties;
+
+import io.awspring.cloud.autoconfigure.AwsClientProperties;
+import io.awspring.cloud.autoconfigure.cloudmap.properties.discovery.CloudMapDiscovery;
+import io.awspring.cloud.autoconfigure.cloudmap.properties.registration.CloudMapRegistryProperties;
+import java.net.URI;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.EnvironmentAware;
+import org.springframework.core.env.Environment;
+
+/**
+ * POJO to capture all cloudmap integration parameters (both registry and discovery).
+ *
+ * @author Hari Ohm Prasath Rajagopal
+ * @since 3.0
+ */
+@ConfigurationProperties(CloudMapProperties.CONFIG_PREFIX)
+public class CloudMapProperties extends AwsClientProperties implements EnvironmentAware {
+
+	/**
+	 * Default cloudmap prefix.
+	 */
+	public static final String CONFIG_PREFIX = "spring.cloud.aws.cloudmap";
+
+	private CloudMapRegistryProperties registry;
```

Comment:
Nested properties classes should be annotated with `@NestedConfigurationProperty` for correct documentation generation.